app-kotlin/build.gradle

```diff
@@ -5,37 +5,46 @@ buildscript {
   }
 
   dependencies {
-    classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.21'
+    classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.0'
   }
 }
 
 apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
 
 android {
-  compileSdkVersion 28
-  buildToolsVersion '28.0.3'
+  compileSdkVersion 31
 
-  defaultConfig {
+    defaultConfig {
     applicationId 'au.com.gridstone.trainingkotlin'
     minSdkVersion 21
-    targetSdkVersion 28
+    targetSdkVersion 31
     versionCode 1
     versionName '1.0'
   }
 
   buildTypes {
     release {
       minifyEnabled false
+      proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
     }
   }
 
-  sourceSets {
-    main.java.srcDirs += 'src/main/kotlin'
+  compileOptions {
+    sourceCompatibility JavaVersion.VERSION_1_8
+    targetCompatibility JavaVersion.VERSION_1_8
+  }
+  kotlinOptions {
```

Comment:
According to [the style guide](https://github.com/Gridstone/KotlinStyleGuide), we need a line break between blocks

Reply:
Thanks from now i will follow style guide